### DISCUSSION TITLE
[QUESTION] Component props and methods unavailable

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxt/nuxt#19094] [Questions] [answered]
Hello, hope you all are having a nice day.

I opened this as a question, because I have the feeling I'm just missing something and that it isn't a bug.

I'm trying to get the methods of a component.
I'm using a `ref` to store a reference to the component of course.

The ref is not null, it returns `{}`, but I don't get any property or method.

I tried to solve it by renaming my component as `.client`.

Obviously  this is happening in an `onMounted` Life Cycle Hook, otherwise the ref is null.

Thanks in advance for any kind response.

**Comment by TheAlexLichter** [comment, marked as answer]
You have to `expose` the methods in the client via [`defineExpose`](https://vuejs.org/api/sfc-script-setup.html#defineexpose). This applies to any Vue.js app using `<script setup>`

**Comment by angelhdzmultimedia** [comment]
### Update

Still not working.

components/test.vue
```vue
<script setup lang="ts">
function clickMe() {
  alert('Ouch!')
}
defineExpose({
  clickMe,
})
</script>

<template>
  <button @click="clickMe">Click Me</button>
</template>

```

app.vue

```vue
<script setup lang="ts">
const test = ref<typeof Test | null>(null)

onMounted(async () => {
  await nextTick(() => {
    alert(JSON.stringify(test.value))
  })
})
</script>

<template>
  <div>
    <Test ref="test" />
  </div>
</template>

```

Tried updating `vue `and `nuxt` to `@latest`.

**Comment by angelhdzmultimedia** [comment]
### Solved
@manniL  yeah your solution works, thanks! I was working on stackblitz.com, something must have been messed up, even after restarting the Nuxt server to refresh component name changes. After closing the browser, now it finally works. 
Thanks for your time, kindness, energy, and solution. Wishing you health, peace, and success!
